### PR TITLE
Add stereo pairing and unpairing operations

### LINKED ISSUE #612: Create stereo pair
Apparently SonoSequencr is able to create a stereo pair from any two speakers (despite the Sonos app not allowing it). Does SoCo expose any stereo pairing functionality, or surround setup functionality today?

### PR DESCRIPTION
**Pull request by pwt**
Closes #612

This PR adds the methods `create_stereo_pair()` and `separate_stereo_pair()` to the SoCo class. This is useful for certain automation scenarios. It also allows pairing of dissimilar Sonos speakers, which is not possible using the Sonos apps. For example, pairing a Sonos One with a Play:1.

Documentation and unit tests are included. 

The implementation is derived from https://github.com/erikmansson/pair-sonos, and I thank @erikmansson for the great work on this.

### CODE REVIEW

**Comment by KennethNielsen** [comment]
Just so we don't forget. There is this feature request on this: #612

File: soco/core.py [hunk 1]
```diff
@@ -1145,6 +1147,43 @@ def unjoin(self):
         zone_group_state_shared_cache.clear()
         self._parse_zone_group_state()
 
+    def create_stereo_pair(self, rh_slave_speaker):
+        """Create a stereo pair.
```
Comment (KennethNielsen) [inline soco/core.py]: New-line between the summary docstring and the general description

File: soco/core.py [hunk 2]
```diff
@@ -1145,6 +1147,43 @@ def unjoin(self):
...
+        # become the master (the left-hand speaker of the pair).
+        # Note that if either speaker is part of a group, the call will
+        # succeed.
+        param = self.uid + ':LF,LF;' + rh_slave_speaker.uid + ':RF,RF'
+        self.deviceProperties.AddBondedZones([
+            ('ChannelMapSet', param)
+        ])
+
+    def separate_stereo_pair(self):
+        """Separate a stereo pair.
```
Comment (KennethNielsen) [inline soco/core.py]: Same here

File: tests/test_core.py [hunk 3]
```diff
@@ -1116,6 +1116,22 @@ def test_soco_set_player_name(self, moco):
                 ('DesiredConfiguration', '')]
         )
 
+    def test_create_stereo_pair(self, moco):
```
Comment (KennethNielsen) [inline tests/test_core.py]: I know that the docstring doverage in the tests is spotty, but please add one

File: tests/test_core.py [hunk 4]
```diff
@@ -1116,6 +1116,22 @@ def test_soco_set_player_name(self, moco):
...
+        moco2 = mock.Mock()
+        moco2.uid = 'RINCON_000XXY1400'
+        moco.create_stereo_pair(moco2)
+        moco.deviceProperties.AddBondedZones.assert_called_once_with(
+            [('ChannelMapSet',
+              'RINCON_000XXX1400:LF,LF;RINCON_000XXY1400:RF,RF')]
+        )
+
+    def test_separate_stereo_pair(self, moco):
```
Comment (KennethNielsen) [inline tests/test_core.py]: and here

File: tests/test_core.py (hunk 3 above)
Comment (pwt) [reply tests/test_core.py]: Calling it 'spotty' is being charitable :)

**Comment by KennethNielsen** [comment]
@pwt great. Ready for merge. I might delay it a tiny bit because #706 changes just about everything and it is just easier to fix a small PR than a big one, but if it is needed I will just do it.

**Comment by pwt** [comment]
OK @KennethNielsen. Thanks.

**Comment by pwt** [comment]
This PR is now Black compliant.

**Comment by KennethNielsen** [comment]
Merged. Thanks for your contribution.

Please drop a note about the change in the release notes thread #709

**Pull request merged**